src/BuildingRegistry/Building/Building.cs

```diff
@@ -50,6 +50,51 @@ public partial class Building : AggregateRootEntity, ISnapshotable
             return newBuilding;
         }
 
+        public void ChangeOutlining(ExtendedWkbGeometry extendedWkbGeometry)
+        {
+            GuardRemovedBuilding();
+
+            if (BuildingGeometry.Method != BuildingGeometryMethod.Outlined)
+            {
+                throw new BuildingHasInvalidBuildingGeometryMethodException();
+            }
+
+            if (BuildingGeometry.Geometry == extendedWkbGeometry)
+            {
+                return;
+            }
+
+            var newBuildingGeometry = new BuildingGeometry(extendedWkbGeometry, BuildingGeometryMethod.Outlined);
+            var plannedOrRealizedBuildingUnits = _buildingUnits.PlannedBuildingUnits
+                .Concat(_buildingUnits.RealizedBuildingUnits)
+                .ToList();
+
+            var buildingUnitsOutsideOfBuildingOutlining = plannedOrRealizedBuildingUnits
+                .Where(x =>
+                    x.BuildingUnitPosition.GeometryMethod == BuildingUnitPositionGeometryMethod.AppointedByAdministrator
+                    && !newBuildingGeometry.Contains(x.BuildingUnitPosition.Geometry));
+
+            if (buildingUnitsOutsideOfBuildingOutlining.Any())
+            {
+                throw new BuildingHasBuildingUnitsOutsideBuildingGeometryException();
+            }
+
+            var buildingUnitsWithPositionDerivedFromBuilding = plannedOrRealizedBuildingUnits
+                .Where(x => x.BuildingUnitPosition.GeometryMethod == BuildingUnitPositionGeometryMethod.DerivedFromObject)
+                .Select(x => x.BuildingUnitPersistentLocalId)
+                .ToList();
+
+            var buildingUnitsPosition = plannedOrRealizedBuildingUnits.Any()
```

Comment:
shouldn't this be `buildingUnitsWithPositionDerivedFromBuilding` ? If so, best make a unit test for it.

Reply:
Correct. The variable buildingUnitsWithPositionDerivedFromBuilding should be used. There's already a test covering that the Center of the new building outline is used. I think the test suffices even though the mistake was not catched https://github.com/Informatievlaanderen/building-registry/blob/849293b02e27da1e5aaf17235544aa444fd2a38b/test/BuildingRegistry.Tests/AggregateTests/WhenChangingBuildingOutline/GivenBuildingExists.cs#L62-L108

Reply:
Changed to use the correct variable.

Reply:
I would make a test that does covers this case. I think if you have a case with one buildingunit (planned/realized) not derived, you can already catch this.

Reply:
Added following test `WithPlannedOrRealizedBuildingUnitsForWhichThePositionIsAppointedByAdministrator_ThenBuildingUnitsPositionRemainsUnchanged`

Now we have:
* ThenBuildingGeometryWasChanged()
* WithSameGeometry_ThenNothing
* WithPlannedOrRealizedBuildingUnitsForWhichThePositionIsDerived_ThenBuildingUnitsPositionWasAlsoChanged()
* WithPlannedOrRealizedBuildingUnitsForWhichThePositionIsAppointedByAdministrator_ThenBuildingUnitsPositionRemainsUnchanged